ClosedXML/Excel/Tables/XLTable.cs

```diff
@@ -860,5 +865,369 @@ internal IXLTable CopyTo(XLWorksheet targetSheet, bool copyData = true)
+            else if (sizeDifference < 0)
+                this.DataRange.Rows(this.LastRow().RowNumber() + sizeDifference + 1, this.LastRow().RowNumber())
+                    .OrderByDescending(r => r.RowNumber())
+                    .ForEach(r => r.Delete());
+
+            this.DataRange.Clear(XLClearOptions.Contents);
+
+            return this.OverwriteTableData(this.DataRange.FirstRow(), data, transpose);
+        }
+
+        public IXLRange ReplaceData(DataTable dataTable)
+        {
+            return ReplaceData(dataTable.Rows.Cast<DataRow>());
+        }
+
+        public IXLRange ReplaceData<T>(IEnumerable<T> data)
+        {
+            if (!data.Any() || data is String)
+                throw new InvalidOperationException("Cannot replace table data with empty enumerable.");
+
+            // Resize table
+            var sizeDifference = data.Count() - this.DataRange.RowCount();
+            if (sizeDifference > 0)
+                this.DataRange.LastRow().InsertRowsBelow(sizeDifference);
+            else if (sizeDifference < 0)
+                this.DataRange.Rows(this.LastRow().RowNumber() + sizeDifference + 1, this.LastRow().RowNumber())
+                    .OrderByDescending(r => r.RowNumber())
+                    .ForEach(r => r.Delete());
+
+            this.DataRange.Clear(XLClearOptions.Contents);
+
+            return this.OverwriteTableData(this.DataRange.FirstRow(), data);
+        }
+
+        internal IXLRange OverwriteTableData(IXLTableRow firstRow, IEnumerable data, bool transpose)
+        {
+            if (data == null || data is String)
+                return null;
+
+            var firstCell = firstRow.FirstCell();
+            var rowNumber = firstCell.Address.RowNumber;
+            var columnNumber = firstCell.Address.ColumnNumber;
+
+            var maxColumnNumber = 0;
+            var maxRowNumber = 0;
+            var isDataTable = false;
+            var isDataReader = false;
+
+            const BindingFlags bindingFlags = BindingFlags.Public | BindingFlags.Instance | BindingFlags.Static;
+            var memberCache = new Dictionary<Type, IEnumerable<MemberInfo>>();
+            var accessorCache = new Dictionary<Type, TypeAccessor>();
+            IEnumerable<MemberInfo> members = null;
+            TypeAccessor accessor = null;
+
+            foreach (var m in data)
+            {
+                var itemType = m.GetType();
+
+                if (transpose)
+                    rowNumber = firstCell.Address.RowNumber;
+                else
+                    columnNumber = firstCell.Address.ColumnNumber;
+
+                if (itemType.IsPrimitive || itemType == typeof(String) || itemType == typeof(DateTime) || itemType.IsNumber())
+                {
+                    Worksheet.SetValue(m, rowNumber, columnNumber);
+
+                    if (transpose)
+                        rowNumber++;
+                    else
+                        columnNumber++;
```

Comment:
You can define this
```
Action increment = () => {
    if (transpose) rowNumber++;
    else columnNumber++;
};
```

And then call `increment();` here and in all places below.